### PR TITLE
serialize figure captions with working references

### PR DESCRIPTION
**Pull request by dylangrandmont**
Requires https://github.com/curvenote/prosemirror-docx/pull/8

### CODE REVIEW

File: src/export/word/schema.ts [hunk 1]
```diff
@@ -116,9 +132,20 @@ export function getNodesAndMarks(styles?: Styles) {
       state.closeBlock(node, styles?.iframe);
     },
     figure(state, node) {
+      const { id, numbered } = node.attrs as Nodes.Figure.Attrs;
+      // TODO: localize this ID
```
Comment (dylangrandmont) [inline src/export/word/schema.ts:136]: Can we actually remove this `TODO` @rowanc1 ? What should I check to confirm whether we need to "localize" this or not? (The cross references appeared to be working)

File: src/export/word/schema.ts [hunk 2]
```diff
@@ -41,6 +41,22 @@ const defaultStyles: Styles = {
...
+  switch (kind) {
+    case CaptionKind.fig:
+      return 'Figure';
+    case CaptionKind.table:
+      return 'Table';
+    case CaptionKind.code:
+      // This is a hack, I don't think word knows about other things!
+      return 'Figure';
+    case CaptionKind.eq:
+      return 'Figure';
```
Comment (rowanc1) [inline src/export/word/schema.ts:54]:
```suggestion
    case CaptionKind.code:
    case CaptionKind.eq:
      // This is a hack, I don't think word knows about other things!
      return 'Figure';
```

File: src/export/word/schema.ts (hunk 1 above)
Comment (rowanc1) [reply src/export/word/schema.ts:136]: We will need an options structure that looks like this:
https://github.com/curvenote/curvenotejs/blob/main/src/export/utils/localizationOptions.ts

The imageSrc isn't needed, but the other ones will be. I think you can actually import these directly from `curvenote`.

**Review by rowanc1 — COMMENTED** [review]
Looks good, I think the todo needs to stay. Minor suggestion for the switch statement.

**Pull request merged**